sigstore/models.py

```diff
@@ -328,6 +335,65 @@ def _verify(self, keyring: RekorKeyring) -> None:
             )
 
 
+class TimestampVerificationData:
+    """
+    Represents a TimestampVerificationData structure.
+    """
+
+    def __init__(self, inner: _TimestampVerificationData) -> None:
+        """Init method."""
+        self._inner = inner
+        self._verify()
+
+    def _verify(self) -> None:
+        """
+        Verifies the TimestampVerificationData.
+
+        It verifies that TimeStamp Responses embedded in the bundle are correctly
+        formed.
+        """
+        signed_ts: list[TimeStampResponse] = []
+        for rfc3161_signed_ts in self._inner.rfc3161_timestamps:
+            try:
+                signed_ts.append(
+                    decode_timestamp_response(rfc3161_signed_ts.signed_timestamp)
+                )
+            except ValueError:
+                raise VerificationError("Invalid Timestamp Response")
+
+        self._signed_ts = signed_ts
+
+    @property
+    def rfc3161_timestamps(self) -> list[TimeStampResponse]:
+        """Returns a list of signed timestamp."""
+        return self._signed_ts
+
+    @classmethod
+    def from_json(cls, raw: str | bytes) -> TimestampVerificationData:
+        """
+        Only used in tests.
+        """
```

Comment:
Does this mean we don't want users to use this method?

Reply:
I've updated the docstring to be more descriptive.